### PR TITLE
front: improve signal edition form

### LINKED ISSUE #5035: Adapt and improve signal edition form generation
## Description

The idea of this issue is to improve the signal edition form.
We are focussing on the logical signals part (the rest will be modified later).

## AC

- Make the  `FlagSignalParam` a check box instead of a dropdown
- Remove input field that contains a constant (`signaling_system` field)
- Rename and translate descriptions
- Reduce the space taken by this part of the form

## Screenshot

Image

### PR DESCRIPTION
**Pull request by SarahBellaha**
Closes #5035

### CODE REVIEW

**Comment by kmer2016** [comment]
Some parts of the form are not translated correctly:
I am using the English language, but in the following screenshot, you can see that the text at the top is in French, while the text below is in English.
image

File: front/src/applications/editor/tools/translationTools.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+      const currentTitleKey = `${defName}.properties.${property}.title`;
+
+      return {
+        ...propertiesAcc,
+        [property]: {
+          ...currentProperty,
+          ...(currentProperty.description && {
+            description: i18n.exists(`infraEditor:${currentDescriptionKey}`)
+              ? t(`infraEditor:${currentDescriptionKey}`)
+              : currentProperty.description,
```
Comment (kmer2016) [inline front/src/applications/editor/tools/translationTools.ts]: We can extract the logic of using the default translation when the key doesn't exist in the translation dictionary and encapsulate it within a function, such as:
```
function translate(key:string, defaultTranslation){
  return i18n.exists(key) ? t(key) : defaultTranslation
}
```
And then you can use it like :
`{ description : translate('infraEditor:${currentDescriptionKey}', currentProperty.description) }`

File: front/src/applications/editor/tools/translationTools.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+          }),
+          ...(currentProperty.title && {
+            title: i18n.exists(`infraEditor:${currentTitleKey}`)
+              ? t(`infraEditor:${currentTitleKey}`)
+              : currentProperty.title,
```
Comment (kmer2016) [inline front/src/applications/editor/tools/translationTools.ts]: same comment here

File: front/src/applications/editor/tools/pointEdition/components.tsx [hunk 3]
```diff
@@ -230,8 +231,52 @@ export const PointEditionLeftPanel: FC<{ type: EditoastType }> = <Entity extends
...
+                is_430: {
+                  'ui:description': ' ',
+                  'ui:widget': CustomFlagSignalCheckbox,
+                },
+              },
+            },
+          },
+        }}
         onSubmit={async (savedEntity) => {
           // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (SharglutDev) [inline front/src/applications/editor/tools/pointEdition/components.tsx]: You should put this line back on top of line 280 to remove the warning :)

File: front/src/applications/editor/tools/pointEdition/components.tsx [hunk 4]
```diff
@@ -230,8 +231,52 @@ export const PointEditionLeftPanel: FC<{ type: EditoastType }> = <Entity extends
...
+                    {
+                      ...savedEntity.properties.logical_signals[0],
+                      signaling_system: 'BAL',
+                    },
+                  ],
+                },
+              };
+            }
+            return savedEntity;
+          };
```
Comment (SharglutDev) [inline front/src/applications/editor/tools/pointEdition/components.tsx:280]: When we check one of these options and save the form, it seems that the change is not taken into account.

File: front/src/applications/editor/tools/pointEdition/CustomFlagSignalCheckbox.tsx [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import { WidgetProps } from '@rjsf/core';
+
+export const CustomFlagSignalCheckbox: React.FC<WidgetProps> = (props) => {
```
Comment (SharglutDev) [inline front/src/applications/editor/tools/pointEdition/CustomFlagSignalCheckbox.tsx:4]: When hovering the checkboxes, there is no "cursor: pointer"

**Review by SharglutDev — CHANGES_REQUESTED** [review]
Thank you for the PR. Left some comments and few translation issues : 

image
If think it should be "Signal" instead of "Signaux" and I remember we agreed not to let the section 
> Cette classe est utilisée pour décrire un signal.


image
In the signal form, there are some not translated fields


image
In the tracksection form, you can remove the "Cette classe..."


image
Same here

File: front/src/applications/editor/tools/translationTools.ts (hunk 1 above)
Comment (SarahBellaha) [reply front/src/applications/editor/tools/translationTools.ts]: Good idea, Thanks !

File: front/src/applications/editor/tools/pointEdition/components.tsx (hunk 4 above)
Comment (SarahBellaha) [reply front/src/applications/editor/tools/pointEdition/components.tsx:280]: The change is saved, but i forgot to give a `checked` attribute to the checkbox to match the value received... Thanks :)

**Review by SharglutDev — CHANGES_REQUESTED** [review]
Thanks for the changes. It seems @kmer2016 first comment is still an issue. When switching language to english, to text remains in french.

**Review by SharglutDev — APPROVED** [review]
Ok great, lgtm, good job :)

File: front/src/applications/editor/tools/translationTools.ts [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+  );
+}
+
+export function translateDefinitions(
+  defsList: { [key: string]: JSONSchema7Definition },
+  t: TFunction
+) {
+  return Object.keys(defsList).reduce(
+    (acc, defName) => {
+      const currentEntity = defsList![defName] as JSONSchema7;
```
Comment (anisometropie) [inline front/src/applications/editor/tools/translationTools.ts]:
```suggestion
  return Object.keys(defsList).reduce<{
    [key: string]: JSONSchema7;
  }>(
    (acc, defName) => {
      const currentEntity = defsList![defName] as JSONSchema7;
```

File: front/src/applications/editor/tools/translationTools.ts [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+              : translate(`infraEditor:${defName}.title`, currentEntity.title, t),
+          }),
+          ...(currentEntity.properties && { properties }),
+        },
+      };
+    },
+    {} as {
+      [key: string]: JSONSchema7;
+    }
+  );
```
Comment (anisometropie) [inline front/src/applications/editor/tools/translationTools.ts]:
```suggestion
    {}
  );
```

File: front/src/applications/editor/tools/translationTools.ts [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+
+export function translateProperties(
+  propertiesList: {
+    [key: string]: JSONSchema7Definition;
+  },
+  defName: string,
+  t: TFunction
+) {
+  return Object.keys(propertiesList).reduce(
+    (propertiesAcc, property) => {
```
Comment (anisometropie) [inline front/src/applications/editor/tools/translationTools.ts]:
```suggestion
  return Object.keys(propertiesList).reduce<{
    [key: string]: JSONSchema7;
  }>(
    (propertiesAcc, property) => {
```

File: front/src/applications/editor/tools/translationTools.ts [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+          ...(currentProperty.title && {
+            title: translate(`infraEditor:${currentTitleKey}`, currentProperty.title, t),
+          }),
+        },
+      };
+    },
+    {} as {
+      [key: string]: JSONSchema7;
+    }
+  );
```
Comment (anisometropie) [inline front/src/applications/editor/tools/translationTools.ts]:
```suggestion
    {}
  );
```

File: front/src/applications/editor/components/EditorForm.tsx [hunk 10]
```diff
@@ -53,6 +57,21 @@ function EditorForm<T extends Omit<EditorEntity, 'objType'> & { objType: string
...
+  const translatedSchemaDefs = schema.$defs && translateDefinitions(schema.$defs, t);
+
+  const translatedSchemaEntity =
+    translatedSchemaDefs && schema.title && translatedSchemaDefs[schema.title];
+
+  const translatedSchema = {
+    ...translatedSchemaEntity,
+    $defs: translatedSchemaDefs,
+    properties:
+      schema.properties && schema.title && translateProperties(schema.properties, schema.title, t),
```
Comment (anisometropie) [inline front/src/applications/editor/components/EditorForm.tsx]: use a memo, you really don’t want to translate all this every single render

**Review by anisometropie — CHANGES_REQUESTED** [review]

File: front/src/applications/editor/tools/pointEdition/components.tsx [hunk 11]
```diff
@@ -230,7 +231,53 @@ export const PointEditionLeftPanel: FC<{ type: EditoastType }> = <Entity extends
...
+              !savedEntity.properties.logical_signals[0].signaling_system
+            ) {
+              return {
+                ...savedEntity,
+                properties: {
+                  ...savedEntity.properties,
+                  logical_signals: [
+                    {
+                      ...savedEntity.properties.logical_signals[0],
+                      signaling_system: 'BAL',
```
Comment (clarani) [inline front/src/applications/editor/tools/pointEdition/components.tsx:273]: Why is it BAL here ?

**Review by clarani — APPROVED** [review]
LGTM, tested :heavy_check_mark:

**Review by anisometropie — APPROVED** [review]

**Pull request merged**